analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/TypeTypeFilter.java

```diff
@@ -8,4 +8,8 @@ public class TypeTypeFilter {
   public boolean equals(Object obj) {
     return obj instanceof TypeType && Objects.equals(TypeType.CLASS, obj);
   }
+  @Override
+  public int hashCode(){
+    return 31 * Objects.hashCode(TypeType.CLASS);
+  }
```

Comment:
I wouldn't expect a `TypeTypeFilter` to be used as a key in a hash map, and if it is, I'm not sure this makes sense as a `hashCode`. It would always return the same value, right? In that case, wouldn't `return 31;` or `return 0;` do the same thing?

Would it perhaps make more sense to throw `UnsupportedOperationException`?

Reply:
This one is the most confusing since (atlesat currently) we have no plans of doing anything with hashcode. I could not really test it since main was broken but now that I can this method doesn't even get called making it useless, so ye I do think it is best to just change it to return 0 just to satisfy sonarCloud. I am not sure what you mean with UnsupportedOperationException